### DISCUSSION TITLE
Intermittent "'Issued at' claim ('iat') must be an Integer" error

### DISCUSSION THREAD

**Discussion by andreiz** [community/community#74745] [Actions]
### Select Topic Area

Bug

### Body

I have an action that uses peter-murray/workflow-application-token-action@v2 to acquire permission tokens from the Github app installed in our org. That particular step intermittently gets and error:

```
Error: Failed to connect as application; status code: 401
'Issued at' claim ('iat') must be an Integer representing the time that the assertion was issued
```

From poking around it seems that this can happen if the system time on the machine doing the API request is out of sync. This action runs on macos-latest-xlarge, so is it possible that these runners have this problem and if so, where do I report it?

**Comment by allwalte** [comment]
Having this same issue (but with the official github token generation action), also only on our jobs on mac runners.  Also occasionally getting "'Expiration time' claim ('exp') is too far in the future"

**Reply by allwalte** [reply]
So I just added in a quick and dirty check:
```
echo 'local:'
  date
  echo 'server:'
  curl -sI https://api.github.com/ | grep -Fi "date" | awk '{ print $6 }'
```
  
And on runs where the token generation worked, those were relatively close together, often a few seconds off but <1m.  But, on one where it failed to generate the token they were about 3 minutes off:
```
local:
Sat Nov 18 00:06:09 UTC 2023
server:
00:03:14
```

**Comment by kianhk6** [comment]
Yeah I am facing the same issue

**Comment by allwalte** [comment]
Apparently it's an acknowledged issue and they're working on a fix - in the meantime the workaround is to set the time from a time server at the start of your action.
